nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/json/JsonPathRowRecordReader.java

```diff
@@ -180,7 +181,7 @@ protected Object convert(final Object value, final DataType dataType, final Stri
             return new MapRecord(childSchema, values);
         }
 
-        if (value instanceof String) {
+        if (value instanceof String && Objects.nonNull(dataType)) {
```

Comment:
To be consistent with the checks done in other parts of the code, would you be ok with?
```suggestion
        if (dataType != null && value instanceof String) {
```

If yes, the Objects import should be removed.

Reply:
👍